event_listener.h

```diff
@@ -98,6 +98,7 @@ namespace EventInfo {
     FXName effect;
     Color color = Color::WHITE;
     Vec2 targetOffset = Vec2(0, 0);
+    bool alwaysShow = false;
```

Comment:
What is this for? To let the player always see where the enemy escaped to?

Reply:
No, this is for TELEPORT_IN: when player teleports outside of current view, fx will be invisible.

Reply:
But you set it to true for targetPos, which is where the creature is moved. Note that this should work when both player and enemies teleport.

Perhaps the right way to do this is:
addEvent [ TELEPORT_OUT ]
moveCreature
addEvent [ TELEPORT_IN ]

(without this extra flag)

Reply:
OK, if moveCreature will update visiblityMap immediately then this will work without alwaysShow flag.